### PR TITLE
Lum 19825 automate bvt setup

### PR DESCRIPTION
**Pull request by Blackbaud-DiHuynh**
@blackbaud/micro-cervezas

### CODE REVIEW

**Comment by Blackbaud-DiHuynh** [comment]
Going to follow up with  a PR for the case the bvtConfiguration already exists

File: src/main/java/com/blackbaud/loc/datasync/kafka/resources/BvtConfigurationResource.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriInfo;
+import java.net.URI;
+
+@Named
+@Path(ResourcePaths.BVT_CONFIGURATION_PATH)
+@Produces(MediaType.APPLICATION_JSON)
+public class BvtConfigurationResource {
+
+    @Inject
```
Comment (Blackbaud-ChristopherCotar) [inline src/main/java/com/blackbaud/loc/datasync/kafka/resources/BvtConfigurationResource.java]: I think we are favoring `@Autowired` over `@Inject` nowadays right?

File: src/main/java/com/blackbaud/loc/datasync/kafka/resources/BvtConfigurationResource.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    @Context
+    private UriInfo uriInfo;
+
+    @POST
+    public Response save(@Valid BvtConfigurationRequest bvtConfigurationRequest) {
+        BvtDataSyncProperties properties = new BvtDataSyncProperties(bvtConfigurationRequest.getBvtName());
+        ClientConfiguration clientConfiguration = properties.toClientConfiguration();
+        ClientConfigurationEntity saved = repository.save(clientConfigurationConverter.toEntity(clientConfiguration));
+        URI self = linkBuilder.buildLink(ClientConfigurationLinkBuilder.METHOD_FIND_ONE, saved, uriInfo);
```
Comment (Blackbaud-ChristopherCotar) [inline src/main/java/com/blackbaud/loc/datasync/kafka/resources/BvtConfigurationResource.java]: somebody's hateful

File: src/main/java/com/blackbaud/loc/datasync/kafka/bvt/BvtDataSyncProperties.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    private int parseSiteId() {
+        return Integer.valueOf(this.bvtName.substring(this.bvtName.length()-4, this.bvtName.length()));
+    }
+
+    public ClientConfiguration toClientConfiguration() {
+        return ClientConfiguration.builder()
+                .siteId(parseSiteId())
+                .operationId(-842)
```
Comment (Blackbaud-ChristopherCotar) [inline src/main/java/com/blackbaud/loc/datasync/kafka/bvt/BvtDataSyncProperties.java]: Probably not needed but we should think about making these magic numbers and strings into constants somewhere...

**Comment by Blackbaud-ChristopherCotar** [comment]
General comment: We should definitely discuss this REST design, it will work for now but it definitely feels weird as is, but I don't have a good recommendation myself.

**Pull request merged**